### PR TITLE
add test parts of day for compliments module

### PR DESCRIPTION
**Pull request by roramirez**

### CODE REVIEW

**Comment by morozgrafix** [comment]
LGTM 👍

File: tests/e2e/modules/compliments_spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			app.start().then(function() { done(); } );
+		});
+
+		afterEach(function (done) {
+			app.stop().then(function() { done(); });
+		});
+
+		it("shows correct compliments for part of day", function () {
+
+			var hour = moment().hour();
```
Comment (fewieden) [inline tests/e2e/modules/compliments_spec.js]: i think this test case is bad, the result will be different based on the time the test is executed.

there should be multiple test cases for each part of the day without the use of momentjs.

**Comment by MichMich** [comment]
Ik think @fewieden has a valid point. Could you look into this, @roramirez?

**Comment by roramirez** [comment]
Yes, I can take a look

File: tests/e2e/modules/compliments_spec.js (hunk 1 above)
Comment (roramirez) [inline tests/e2e/modules/compliments_spec.js]: @fewieden Why dont use momentjs?

You mean,  create  3 test case and check if is the part of day accord reflect output module content

Comment (fewieden) [inline tests/e2e/modules/compliments_spec.js]: exactly at least 3 test cases, there can be even more to test the boundary values

**Comment by fewieden** [comment]
hey @roramirez i think you misunderstood what i meant with the momentjs part, because now you will only run one of those tests based on the time of the day as well. you have to hardcode the hour for each test so each test gets executed and you can check if its working correctly for each part of the day at once

**Comment by morozgrafix** [comment]
Sorry, I'm a bit confused. My understanding that this is end to end test that makes sure that correct message is displayed in electron at runtime. Since we can't control at what time Travis will be running it initial commit was covering any of 3 combinations. (Can we force set system time on Travis to simulate specific hour? I'm kind of not sure if that's a good idea anyway). I agree that we don't need to rely on moment.js and can do tests with pure JS Date function. Also agree that splitting it into 3 tests knowing that only one will run at any given point is not ideal. Hardcoding hours wouldn't work, since MM running in electron will pickup system time. Am I missing something?

File: tests/e2e/modules/compliments_spec.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+const Application = require("spectron").Application;
+const path = require("path");
+const chai = require("chai");
+const expect = chai.expect;
+const chaiAsPromised = require("chai-as-promised");
+const moment = require("../../../vendor/moment/moment-with-locales.js");
```
Comment (morozgrafix) [inline tests/e2e/modules/compliments_spec.js]: probably no longer need moment.js

**Comment by roramirez** [comment]
Force the system date to check a test I think is a wrong way.  We need create a test for expected values and behavior on a test system.

The tests for date and time is a complicated are because this run a host system and at a determinated moment.

We can create a new tests for case and workflow mention by @fewieden but these need be a new tests cases. I'm not sure if possible set a date enviroment where run the test... for this we need a method to force and set date on enviroment for test case.

File: tests/e2e/modules/compliments_spec.js (hunk 2 above)
Comment (roramirez) [inline tests/e2e/modules/compliments_spec.js]: Removed

**Comment by MichMich** [comment]
I've merged this for now. We can always improve this if we want.

**Comment by MichMich** [comment]
It seems Travis is currently having issues with these tests. @roramirez Can you look into this?

**Comment by roramirez** [comment]
Yes, I can take a look.  @MichMich Can you tell me what build is failed for this test? . I've seen the Pull Request in Travis but I'dont see nothing related there.

**Comment by morozgrafix** [comment]
@roramirez looks like it was hitting timouts https://travis-ci.org/MichMich/MagicMirror/jobs/199172700

and this one wasn't able to locate and element: https://travis-ci.org/MichMich/MagicMirror/jobs/199173082 (that was one of my tests. I don't think it failed in the past)

**Comment by roramirez** [comment]
@morozgrafix Its so stranger, now are passing the tests

**Comment by qistoph** [comment]
I've seen it on Travis before with my own branch of MM. 
Maybe it happens when their server(s) are a bit busy and the electron app is taking longer to load. 

Is it an idea to keep track (URLs) of failed builds that shouldn't fail on the Trello board, so that we can look into it later with a collection of evidence?

**Comment by roramirez** [comment]
Yes, is good idea @qistoph.  Can you add this there on Trello?

**Comment by qistoph** [comment]
https://trello.com/c/nIrPAN3o/23-fix-failing-tests-on-travis

**Comment by roramirez** [comment]
Thanks you!
